docs/guides/implementation-guides/how-to-implement-dependency-injection.md

```diff
@@ -3,13 +3,127 @@ id: how-to-implement-dependency-injection
 title: How To Implement Dependency Injection
 ---
 
-import MvvmArchitectureDiagram from '/img/gitbook-import/assets/image (3) (1) (2).png';
+# How To Implement Dependency Injection in Avalonia
 
-# How To Implement Dependency Injection
-
-Content in preparation.
+[Dependency injection (DI)](https://en.wikipedia.org/wiki/Dependency_injection) allows developers to write cleaner, more modular, and testable code. It accomplishes this by creating discrete services that are passed around/created as needed. It is especially useful when you are developing multi-platform applications that need specific service implementations for each platform.
 
 This guide will show you how to use Dependency Injection (DI) with _Avalonia UI_ and the MVVM pattern. 
 
-<img src={MvvmArchitectureDiagram} alt=""/>
+Let's assume that your MainViewModel has a dependency on MyService. The constructor for ViewModel should looks like this:
+
+```csharp
+private readonly MyService _myService;
+
+public MapsViewModel(MyService myService)
+{
+    _myService = myService;
+}
+```
+These are the steps you need to do in order to resolve that dependencu using DI.
+
+## Step 1: Install the NuGet package for DI
```

Comment:
Our first step and option should be on using DI without any IoC.
I.e. simple:
```csharp
var window = new MainWindow
{
    DataContext = new MapsViewModel(new MyService())
}
```
Or something like that. It also will serve a good example **why** people prefer IoC containers, to reduce boilerplate. 
All of the IoC libraries will get quickly opinionated.

Reply:
Agree, I kept the example very simple to focus on the DI concept. I could add this in the step one but I would need to pull the whole app class to keep the context...which may be a bit overkill don't you think?

From my understanding, the idea here in this guide is to explain how install the nugget, what code to add and what to configure to have a skeleton of DI working. Am I correct?

Reply:
After a bit more thought I've added your suggestion to the step 1. Could you check If I have captured and articulated your point correctly please?

Reply:
I don't see changes in the step 1?

> From my understanding, the idea here in this guide is to explain how install the nugget

I would say, the idea should be to show options that users have to setup DI, referencing some links. And MS.DI as a practical example with simple steps.

Reply:
Also see https://github.com/AvaloniaUI/avalonia-docs/pull/359#discussion_r1493882114
And https://github.com/AvaloniaUI/avalonia-docs/pull/359#discussion_r1493880450